zerver/tests/test_users.py

```diff
@@ -67,6 +68,21 @@ def test_do_change_is_admin(self) -> None:
         with self.assertRaises(AssertionError):
             do_change_is_admin(user_profile, True, permission='totally-not-valid-perm')
 
+    def test_change_is_primary_admin(self) -> None:
+        iago = self.example_user('iago')
+        hamlet = self.example_user('hamlet')
+
+        self.assertEqual(hamlet.is_primary_admin, False)
+        do_change_is_primary_admin(hamlet, True)
+        self.assertEqual(hamlet.is_primary_admin, True)
+        self.assertEqual(hamlet.is_realm_admin, True)
+
+        self.login(hamlet.email)
+        req = dict(is_primary_admin=ujson.dumps(True))
+        self.assertFalse(iago.is_primary_admin)
+        result = self.client_patch('/json/users/{}'.format(iago.id), req)
+        self.assert_json_success(result)
```

Comment:
Also, a test like this should never end with just checking success -- we should also refetch Iago from the database and verify his state after it was supposedly changed in an API call.